recommender/recommender.py

```diff
@@ -657,43 +678,106 @@ def endorse_resource(self, data, _suffix=''):
     @XBlock.json_handler
     def deendorse_resource(self, data, _suffix=''):
         """
-        Deendorse an entry of resource.
+        Remove an entry of resource.
 
         Args:
                 data: dict in JSON format
-                data['id']: the ID of the resouce to be deendorsed
-                data['reason']: the reason why the resouce was deendorsed
+                data['id']: the ID of the resouce to be removed
+                data['reason']: the reason why the resouce was removed
         Returns:
                 result: dict in JSON format
-                result['Success']: the boolean indicator for whether the deendorsement is complete
-                result['error']: the error message generated when the deendorsement fails
-                result['recommendation']: (Dict) the deendorsed resource
-                result['recommendation']['reason']: the reason why the resouce was deendorsed
+                result['Success']: the boolean indicator for whether the removal is complete
+                result['error']: the error message generated when the removal fails
+                result['recommendation']: (Dict) the removed resource
+                result['recommendation']['reason']: the reason why the resouce was removed
         """
-        # TODO: this function was named delete_resource previously, which should be changed in test_recommender
         if not self.get_user_is_staff():
-            msg = 'Deendorse resource without permission'
+            msg = 'Remove resource without permission'
```

Comment:
Are you trying to indicate that the user does not have permission to remove the resource? Or that the server does not have permission to remove the resource? What is the user is not currently authenticated -- shouldn't the server indicate that the user should authenticate first, because they may be authorized to remove the resource after they are authenticated?